src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,1077 @@
+
+        # Clean up complete.status files
+        self.runtime.env_layer.file_system.delete_files_from_dir(self.runtime.status_handler.status_file_path, '*.complete.status')
+
+    def test_only_assessment_packages_truncation_under_size_limit(self):
+        """ Perform no truncation on assessment packages list.
+        Expecting:
+        assessment substatus status: success,
+        no assessment tombstone records,
+        assessment errors code: 0 (success),
+        assessment errors details code: 0 (success). """
+
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        patch_count = 500
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        # Assert complete status file size < 128kb
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.__assert_patch_summary_from_status(substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS)
+        self.__asert_message_json_from_status(substatus_file_data[0]["status"]["substatus"][0], patch_count, error_count=0)
+
+        # Assert no truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        message = json.loads(substatus_file_data[0]["status"]["substatus"][0]["formattedMessage"]["message"])
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.assertEqual(substatus_file_data[0]["status"]["substatus"][0]["name"], Constants.PATCH_ASSESSMENT_SUMMARY, )
+        self.assertEqual(substatus_file_data[0]["status"]["substatus"][0]["status"], Constants.STATUS_SUCCESS.lower())
+        self.assertEqual(len(message["patches"]), patch_count)
+        self.assertNotEqual(message["patches"][-1]['patchId'], "Truncated_patch_list_id")
+        self.assertTrue('Truncated_patch_list_id' not in message["patches"][-1]['name'])
+        self.assertEqual(message["errors"]["code"], 0)
+        self.assertEqual(len(message["errors"]["details"]), 0)
+        self.assertFalse("review this log file on the machine" in message)
```

Comment:
All of this and maybe other code is repeated 4 times. Counts and repeats vary. There was an open ask to deduplicate all test code like this. "It does seem like a lot of tests are just copies of an test written first with minor tweaks - this is a sign that test refactoring is significantly possible to allow for long-term maintainability."

Reply:
these assertions are for no truncation logic it's different than when truncation is apply, I've extracted them into a function and use it in the 2 non-truncation utests. pending for review